pkg/kubectl/cmd/get/get_test.go

```diff
@@ -1370,30 +1521,56 @@ func TestWatchOnlyResource(t *testing.T) {
 			switch req.URL.Path {
 			case "/namespaces/test/pods/foo":
 				return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: cmdtesting.ObjBody(codec, &pods[1])}, nil
-			case "/namespaces/test/pods":
-				if req.URL.Query().Get("watch") == "true" && req.URL.Query().Get("fieldSelector") == "metadata.name=foo" {
-					return &http.Response{StatusCode: 200, Header: cmdtesting.DefaultHeader(), Body: watchBody(codec, events[1:])}, nil
-				}
-				t.Fatalf("request url: %#v,and request: %#v", req.URL, req)
-				return nil, nil
 			default:
-				t.Fatalf("request url: %#v,and request: %#v", req.URL, req)
+				t.Errorf("request url: %#v,and request: %#v", req.URL, req)
 				return nil, nil
```

Comment:
After change to `t.Errorf` you should return error, no?

Reply:
here I should because the fake doesn't understand the request